quai/api_backend.go

```diff
@@ -244,7 +244,16 @@ func (b *QuaiAPIBackend) GetBloom(hash common.Hash) (*types.Bloom, error) {
 	if nodeCtx != common.ZONE_CTX {
 		return nil, errors.New("getBloom can only be called in zone chain")
 	}
-	return b.quai.core.Slice().HeaderChain().GetBloom(hash)
+	bloom, err := b.quai.core.Slice().HeaderChain().GetBloom(hash)
```

Comment:
you should have this logic inside the headerchain method, not here in api_backend